include/seqan3/std/bit

```diff
@@ -49,3 +72,156 @@ enum class endian
 } //namespace std
 
 #endif // __cpp_lib_endian
+
+#if !defined(SEQAN3_CPP_LIB_INT_POW2) || !defined(SEQAN3_CPP_LIB_BITOPS)
+
+namespace std::detail
+{
+/*!\brief How many bits has a type?
+ * \ingroup bit
+ * \tparam type_t The type to determine the number of bits.
+ */
+template <typename type_t>
+constexpr auto sizeof_bits = CHAR_BIT * sizeof(type_t);
+} // namespace std::detail
+
+#endif // !defined(SEQAN3_CPP_LIB_INT_POW2) || !defined(SEQAN3_CPP_LIB_BITOPS)
+
+#ifndef SEQAN3_CPP_LIB_INT_POW2
+
+namespace std
+{
+
+namespace
+{
```

Comment:
why do we need the anonymous namespace?
Can you add a short comment?

Reply:
I'll add a comment.

This basically allows us to always define these functions regardless whether the functions are already defined or not, because functions within anonymous namespaces have less priority than functions directly within the namespace.

Example:
https://godbolt.org/z/s6hPsM

I'm not sure how I can explain that in a single, short sentence.

Reply:
> That means some functions may be defined?
> 
> regarding the comment, I meant why do we need the namespace, not necessarily what it does.

I'm not completely sure what you mean. The main idea is to always define `std::<anonymous namspace>::popcount`, if `std::popcount` is defined it will be preferred, otherwise `std::<anonymous namspace>::popcount` will be used.